src/deep-resource/lib/Resource/Request.js

```diff
@@ -364,34 +351,147 @@ export class Request {
         let publicCache = cache.shared;
         let publicCacheKey = publicCache.buildKeyFromRequest(this);
 
-        this.loadResponseFromCache(publicCache, publicCacheKey, (error, response) => {
+        this._loadResponseFromCache(publicCache, publicCacheKey, (error, response) => {
           if (!error) {
             callback(response);
-
             return;
           }
 
-          this._send(decoratedCallback);
+          this._send(callback);
         });
 
-        return;
+      } else {
+        this._send(callback);
       }
-
-      this._send(decoratedCallback);
     });
 
     return this;
   }
 
   /**
+   * @param {Function} callback
+   * @returns {Request}
+   */
+  _send(callback = () => {}) {
+    let logService = this.action.resource.log;
+    let event = {
+      service: "deep-resource",
+      resourceType: "Browser",
+      resourceId: this.native ? this.action.source.original : this.action.source.api,
+      eventName: this.method,
+      eventId: this.customId,
+      requestId: this.customId,
+      payload: this.payload,
+    };
+
+    logService.rumLog(event);
+
+    let decoratedCallback = (response) => {
+      this._saveResponseToCache(response, (error, data) => {
+        if (error) throw error;
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Expected '{' and instead saw 'throw'.](https://www.codacy.com/app/MitocGroup/deep-framework/file/2034169663/issues/source?bid=2895475&fileBranchId=3129256#l391)